pkg/reconciler/adapter.go

```diff
@@ -146,7 +146,11 @@ func commonAdapterDeploymentOptions(rcl v1alpha1.Reconcilable) []resource.Object
 		}
 	}
 
-	return objectOptions
+	overrides := rcl.GetAdapterOverrides()
+	if overrides == nil {
+		return objectOptions
+	}
+	return append(objectOptions, adapterOverrideOptions(overrides))
```

Comment:
To echo one of my previous comments, if we were to make this behavior optional (and I think it logically makes sense to consider it optional), that logic could look like:

```golang
	if cfbl, canConfigureAdapter := rcl.(v1alpha1.AdapterConfigurable); canConfigureAdapter {
		if overrides := cfbl.GetAdapterOverrides(); overrides != nil {
				return append(objectOptions, adapterOverrideOptions(overrides))
		}
	}

	return objectOptions
```